libraries/resource_providers/image_streamer/api300/artifact_bundle_provider.rb

```diff
@@ -66,6 +66,7 @@ def upload
 
         def extract
           raise "#{@resource_name} #{@name} does not exist in the appliance. Cannot run extract action." unless @item.exists?
+          @item.retrieve! || raise("#{@resource_name} '#{@name}' not found!")
```

Comment:
Not sure whats the use or OR operator here, cant we add "unless" to check if that particular resource's name exist in OV or not and raise exception accordingly?